src/metrics_server.rs

```diff
@@ -0,0 +1,38 @@
+use actix_web::http::header::CONTENT_TYPE;
+use actix_web::{get, HttpResponse, Responder};
+use lazy_static::lazy_static;
+use prometheus::{
+    register_histogram, register_int_counter, Encoder, Histogram, IntCounter, TextEncoder,
+};
+
+lazy_static! {
+    static ref METRICS_REQ_TOTAL: IntCounter = register_int_counter!(
+        "metrics_requests_total",
```

Comment:
Are all of those keys namespaced? If they aren't, we should prefix them with `xenos_`